### DISCUSSION TITLE
Would you consider adding a Machine-readable progress bar option?

### DISCUSSION THREAD

**Discussion by joeyballentine** [Textualize/rich#3000] [Ideas]
Right now, rich disables progress reporting when something that uses rich is run under a subprocess. This is really inconvenient if you want to have the parent program know how far along the subprocess is.

I implemented a machine-readable progress feature for pip (https://github.com/pypa/pip/pull/12084) and they suggested I ask if this is instead something rich should handle. 

Is this something you'd be willing to support?

**Comment by willmcgugan** [comment]
I'd certainly consider it. If feels like there should be hooks for this, and it should be left up to the developer to decide on the format.
